### ISSUE TITLE
Uninstall of second Gantry 5 template leaves residual menu assignment

### ISSUE THREAD

**Issue by DanGthatRokGallery** [gantry/gantry5#537]
Using Ethereal as your installation template install Callisto or Ambrosia to that. I find it pesky that the new install sets itself up as the Default taking that away from Ethereal.

However the real issue is if you now uninstall Ambrosia and look at the menu Assignments in your Ethereal template - you will find the Ambrosia Theme -> Home page as an existing option.

Reported here by the user:
http://www.rockettheme.com/forum/joomla-template-ambrosia/244366-tried-to-switch-from-ethereal-to-ambrosia

and then I duplicated the issue on my local server.

**Comment by w00fz** [comment]
I don't like it either that we set the new installed template as default as well as creating and setting as default a new Menu.

**Comment by gyvermac** [comment]
Yes I think that install shouldn't modify any assignment neither create any menu...

**Comment by DanGthatRokGallery** [comment]
I know this will NEVER happen but some designer installs a pretty new template on a production site and and won't take long before the client and their S**T hits the fan :(

**Comment by mahagr** [comment]
There's really nothing we can do on uninstall as Joomla doesn't support install scripts for the templates on uninstall (until it seems.. I fix it for Joomla).

I think this is really something to @rhukster to decide, as the current implementation was supposed to be optional step after the installation until we talked that people would not know what to do with the template if we didn't make it to work out of the box.

I agree with you partially.. I still believe that it would be useful to have pre-build sample data for the development sites (without needing to start from scratch by installing RL), but I do think it's better to be an optional step after installation.

**Comment by JoomFX** [comment]
I also find creating a menu item and setting the template as default a bit strange.
It's just that none of the other templates and frameworks do it and the users are not used to it.

**Comment by kat05** [comment]
i am not sure about the title of this issue and it being marked as a bug?

> Uninstall of second Gantry 5 template leaves residual menu ASSIGNMENT

there is no 'menu assignment' after uninstalling the ambrosia template.

yes, the 'ambrosia theme' menu that has been added during the ambrosia template installation is still present in the menu manager, therefor it is the correct behavior that it shows in the ethereal's menu assignments as an available option. but there are no actual pages assigned to it, at least not in my testing. with the current way gantry handles templates installations and menu creation upon install, for me this is the EXPECTED BEHAVIOR.

after you manually delete the 'ambrosia theme' menu from the menu manager, this is no longer showing in the ethereal's menu assignments, also that is the expected and correct behavior at this point.

HOWEVER, i agree that this behavior can be very confusing to users, maybe we just need to point this out more in the documentation if the behavior continues to be like it is now? 

make them aware that the template install
a) adds a new menu
b) changes the default menu item and 
c) sets the new template as the default template.

all of these changes will have to be manually reverted by users if they decide to remove the template again. if they do, everything should be back to what it originally was. if it's NOT, THEN i would consider that a BUG.

**Comment by kat05** [comment]
thinking about it more, after a new gantry 5 theme has been installed, why not have a message there informing the member:

1) the ambrosia template has been installed successfully!
2) the ambrosia theme menu has been added to your menu manager successfully!
3) the ambrosia template has been set as your default template!
4) the ambrosia theme menu home item has been set as your default menu item!
INFO: please review the above changes to your site. in order to undo the above changes, please refer to our documentation here (LINK)

**Comment by rhukster** [comment]
The intention of setting the template by default on install would save a step that 99% of the time you would do anyway.  If it's causing too much confusion, i'm totally cool with not doing that and letting people manually activate it.

**Comment by mahagr** [comment]
Leaving data behind is a Joomla bug

**Comment by mahagr** [comment]
Sorry, clicked on wrong button. I agree that leaving data behind is a bug; but it's really a Joomla bug (uninstall script not called on templates).

Here is what I would do: have a button to set up menu, letting users to do it manually as a part of installation process.

**Comment by rhukster** [comment]
Sounds like a good solution!

**Comment by w00fz** [comment]
This almost calls again for that post installation hook I brought up a while back, where after installation you would be taken to a G5 view in admin where things happen.
SCSS gets (re)compiled, cache gets cleared, anything that needs to be initialized/checked happens and so it could also offer to a) set the template as default b) install the new menu and set it as default.

**Comment by mahagr** [comment]
Yes, which is also why I brought it up. I really want it rather soon than later. :)

**Comment by w00fz** [comment]
I set this to 5.0.1 for now. Something we might want to look at after release/wordpress.
Definitely an enhancement rather than a bug here anyway.

**Comment by mahagr** [comment]
Moved to future releases because of it really needs a fix in Joomla.

**Comment by mahagr** [comment]
As of Joomla 3.7.2, there's still no way to run scripts in Joomla template uninstaller.
